src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -428,9 +424,16 @@ protected virtual void Initialize()
 
                 LoadCustomExtensions();
 
+                // Do this after we've loaded the custom extensions. That gives an extension an opportunity to plug in their own implementations.
+                if (storageString != null)
+                {
+                    var lockManager = this.ScriptConfig.HostConfig.GetService<IDistributedLockManager>();
+                    blobManagerCreation = BlobLeaseManager.Create(lockManager, TimeSpan.FromSeconds(15), hostConfig.HostId, InstanceId, TraceWriter, hostConfig.LoggerFactory);
+                }
+
                 // Create the lease manager that will keep handle the primary host blob lease acquisition and renewal
                 // and subscribe for change notifications.
-                _blobLeaseManager = blobManagerCreation.GetAwaiter().GetResult();
+                _blobLeaseManager = blobManagerCreation;
```

Comment:
nit: With this implementation, we no longer have the need to assign it to a temporary variable here as we're not kicking the creation process in the background and getting the result later, so we can get rid of it.